app/services/pii/cacher.rb

```diff
@@ -0,0 +1,31 @@
+module Pii
+  class Cacher
+    def initialize(user, user_session)
+      @user = user
+      @user_session = user_session
+    end
+
+    def save(password, profile = user.active_profile)
+      return unless profile
+      decrypted_pii = profile.decrypt_pii(password)
+      pii_json = decrypted_pii.to_json
+      encrypted_pii = encryptor.encrypt(pii_json)
+      user_session[:encrypted_pii] = encrypted_pii
+    end
```

Comment:
Ah, yes. I keep forgetting about that. I think some comments at the top of this class explaining why it's needed would be helpful.